api/reviews_test.go

```diff
@@ -568,7 +587,260 @@ func TestAddReviewHandler(t *testing.T) {
 			api.addReviewHandler(response, request)
 			Convey("Then the HTTP response code is 400", func() {
 				So(response.Code, ShouldEqual, http.StatusBadRequest)
-				So(response.Body.String(), ShouldEqual, "invalid review. Missing required field\n")
+				So(response.Body.String(), ShouldEqual, "invalid review\n")
+			})
+		})
+	})
+}
+
+func TestUpdateReviewHandler(t *testing.T) {
+	Convey("Given an HTTP PUT request to the /books/{id}/reviews/{review_id} endpoint", t, func() {
+
+		Convey("When the book and review exist, and the review update is valid", func() {
+			// TODO - how can I get GetReviewFunc to return a different value depending on how many times it has been called?
```

Comment:
This gets a bit messy but  if you keep track/maintain a call count you could create a func like the example below and return different responses determined by what the invocation count is.

Over simplifier example but hopefully illustrates my point.
https://play.golang.org/p/ckZkPJ5DeYS